AzFilesHybrid/AzFilesHybrid.psm1

```diff
@@ -4041,7 +4014,7 @@ function Debug-AzStorageAccountAuth {
                 $checksExecuted += 1
                 Write-Verbose "CheckUserRbacAssignment - START"
 
-                Request-ConnectAzureAD
+                Request-ConnectMsGraph -Scopes "User.Read.All", "GroupMember.Read.All"

```

Comment:
This is very nice 👍 Actually a security gain, as you now only get a token with the minimal access.

One concern I have is: for commands that use a lot of other commands (e.g., the debug command), will the user be prompted multiple times to log in with various scopes? If so, should we maybe add a Request-ConnectMsGraph to the debug command that preemptively gets a token with all scopes it needs?

Reply:
The behaviour with this change now is that, when cx run Debug, it will pop up the grant window only once, and ask for grants for User.Read.All and Group.Member.ReadAll. after they granted, the following MS calls dont need grants anymore since all the tests in Debug only needs these 2 permissions.

And I think for the scope of the Debug command, we prob only need this 2 permissions for now.

Reply:
Nice. This is a nitpick, but we may want to add a comment explaining this above the Request-ConnectMsGraph line.